app/controllers/users_controller.rb

```diff
@@ -26,4 +31,21 @@ def show
   def user_params
     params.require(:user).permit(:email, :password, :full_name)
   end
+
+  def is_token_invalid?
+    params[:invite_token].present? && !@inviter
+  end
+
+  def set_invitation_datas
+    @invite_user = InviteUser.find_by(token: params[:invite_token])
+    @inviter = @invite_user.try(:inviter)
+  end
+
+  def build_relationship_with_inviter
+    if @inviter
+      @user.follow(@inviter)
+      @inviter.follow(@user)
+      @invite_user.destroy
```

Comment:
Most of the time you don't want to destroy anything. In the future you may want to be able to know how many invitations have been send out on your app. If you destroy them, you won't know. You would usually mark an object as deleted in the db, but not actually delete it.